public/tracker/modules/task/directives/taskHistory/taskHistory.js

```diff
@@ -0,0 +1,84 @@
+angular
+    .module('Tracker')
+
+    .directive('taskHistory', function () {
+            return {
+                restrict: 'A',
+                templateUrl: 'tracker/modules/task/directives/taskHistory/taskHistory.html',
+                controller: function ($scope,
+                                      Task,
+                                      TaskHistory,
+                                      historyInfoService) {
+                    var allMessages;
+
+                    $scope.getInclude = function (messageType) {
+                        var historyType = _.find(historyInfoService, function (historyType) {
+                            return historyType.type === messageType;
+                        });
+                        return historyType.templateUrl;
+                    };
+
+                    function initComment() {
+                        $scope.comment = new TaskHistory({
+                            text: ''
+                        });
+                        loadHistory();
+                    }
+
+                    $scope.$watch('task', function (task) {
+                        if (task) {
+                            initComment();
+                        }
+                    });
+
+                    $scope.saveComment = function ($event) {
+                        if ($event.keyCode == 13 && $event.shiftKey) {
+                            createComment();
+                        }
+                    };
+
+                    function createComment() {
+                        $scope.comment.$save({taskId: $scope.task._id, nested: 'comments'}, function () {
+                            updateCommentCounter();
+                            loadHistory();
+                        });
+                    }
+
+                    function updateCommentCounter() {
+                        $scope.task.commentsCounter++;
+                        $scope.task.$update({taskId: $scope.task._id});
```

Comment:
this is responsibility of server side, it ($scope.task.commentsCounter++;) can be easy hacked